### PR TITLE
Add instructions for building Swift on Windows using clang-cl/MSVC

### PR DESCRIPTION
**Pull request by hughbe**
Extracted from #5904

### CODE REVIEW

File: docs/Windows.md [hunk 1]
```diff
@@ -38,7 +38,117 @@ Additionally, the ICU headers and libraries need to be provided for the build.
...
+```
+
+### 7. Build Swift
+- This must be done from within a developer command prompt, and could take up to 2 hours.
+- You may need to adjust the SWIFT_WINDOWS_LIB_DIRECTORY parameter depending on your target platform or Windows SDK version.
+```
+mkdir "%swift_source_dir%/build/Ninja-DebugAssert/swift-windows-amd64/ninja"
+pushd "%swift_source_dir%/build/Ninja-DebugAssert/swift-windows-amd64/ninja"
+cmake -G "Ninja" "%swift_source_dir%/swift"^
+ -DCMAKE_CXX_FLAGS="/std:c++14"^
```
Comment (compnerd) [inline docs/Windows.md]: We dont use C++14 features yet, why do we have to push `/std:c++14`?

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@hughbe are you okay with this going in before the rest of the patches?  (I dont have a strong preference).

**Comment by hughbe** [comment]
Fine by me - order doesn't matter for any of the patches. couple of changes might need to be made though, e.g remove libedit, get rid of c++ 14 flag.

File: docs/Windows.md (hunk 1 above)
Comment (hughbe) [inline docs/Windows.md]: This is redundant, and snuck into the docs

Comment (hughbe) [inline docs/Windows.md]: I've removed this, and also got rid of the mentions of libedit (the cmake flag to disable it no longer exists)

File: docs/Windows.md [hunk 2]
```diff
@@ -38,7 +38,115 @@ Additionally, the ICU headers and libraries need to be provided for the build.
 ```
 
 ## MSVC
-To be filled in.
+- Windows doesn't currently have a build script. You'll need to run commands manually to build Swift on Windows.
+- Release/RelWithDebInfo modes have not been tested and may not be supported.
+- x86 has not been tested.
```
Comment (compnerd) [inline docs/Windows.md]: FYI, the cross-compilation based windows support has been tested and shown to be working :-).

File: docs/Windows.md [hunk 3]
```diff
@@ -38,7 +38,115 @@ Additionally, the ICU headers and libraries need to be provided for the build.
...
+ -DSWIFT_PATH_TO_LLVM_SOURCE="%swift_source_dir%/llvm"^
+ -DSWIFT_PATH_TO_LLVM_BUILD="%swift_source_dir%/build/Ninja-DebugAssert/llvm-windows-amd64"^
+ -DSWIFT_PATH_TO_CLANG_SOURCE="%swift_source_dir%/llvm/tools/clang"^
+ -DSWIFT_PATH_TO_CLANG_BUILD="%swift_source_dir%/build/Ninja-DebugAssert/llvm-windows-amd64"^
+ -DICU_UC_INCLUDE_DIRS="%swift_source_dir%/icu/include"^
+ -DICU_UC_LIBRARY_DIRS="%swift_source_dir%/icu/lib64"^
+ -DICU_I18N_INCLUDE_DIRS="%swift_source_dir%/icu/include"^
+ -DICU_I18N_LIBRARY_DIRS="%swift_source_dir%/icu/lib64"^
+ -DICU_UC_LIB_NAME="icuucd"^
+ -DICU_I18N_LIB_NAME="icuind"^
```
Comment (compnerd) [inline docs/Windows.md]: Is there a particular reason to use a debug build of ICU instead of a release ICU build as it is an external dependency?

**Comment by compnerd** [comment]
@swift-ci please test

File: docs/Windows.md (hunk 2 above)
Comment (hughbe) [inline docs/Windows.md]: Awesome! I meant a from-windows build (not X compilation) 

I know I'm having some problems with release mode not working, though - I hit an assertition compiling the stdlib

File: docs/Windows.md (hunk 3 above)
Comment (hughbe) [inline docs/Windows.md]: Nope, just the one I've tested.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**